File: 81.jl
countries = readcsv("countries.csv")
splitted_names = filter(x -> length(x) > 0, split.(countries[2:end,2]))
splitted_names = map(x -> lowercase.(x), splitted_names)

open("80.py.out") do f
    for line in readlines(f)
        if length(line) < 1
            continue
        end
        words = filter(x -> !isempty(x), lowercase.(split(line[1:end])))
        newwords = String[]
        idx = 1
        while idx < length(words)
            word = words[idx]
            found = false
            for c in splitted_names
                cl = length(c)
                if idx + cl - 1 >= length(words)
                    break
                end
                if length(c) > 1 && words[idx:idx+cl-1] == c
                    push!(newwords, join(c, "_"))
                    idx = idx + cl
                    found = true
                end
            end
            if !found
                idx += 1
                push!(newwords, word)
            end
        end

        words = newwords
        if length(words) > 0
            for token in words
                print(token, " ")
            end
            println("")
        end
    end
end

File: 81.py
import pandas as pd
import sys
import numpy as np

countries = pd.read_csv("countries.csv")
names = np.array(countries["ISO 3166-1に於ける英語名"])
splitted_names = []
for name in names:
    s = name.split(" ")
    s = list(filter(lambda x: x != "", map(lambda b: b.strip(), s)))
    splitted_names.append(s)
if False:
    for name in splitted_names:
        print(name)

with open("80.py.out") as f:
    for line in f.readlines():
        words = line[:-1].split(" ")
        words = list(filter(lambda x: x != "", words))
        newwords = []
        idx = 0
        while idx < len(words):
            word = words[idx]
            found = False
            for c in splitted_names:
                cl = len(c)
                if idx + cl >= len(words):
                    break
                if len(c) > 1 and words[idx:idx + cl] == c:
                    newwords.append("_".join(c))
                    idx = idx + cl
                    found = True
            if not found:
                idx += 1
                newwords.append(word)

        words = newwords
        if len(words) > 0:
            for token in words:
                sys.stdout.write("{} ".format(token))
            sys.stdout.write("\n")

sys.exit(0)
